contracts/client/ClientFrontend.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity ^0.5.15;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { Tx } from "../libs/Tx.sol";
+import { Types } from "../libs/Types.sol";
+import { Transition } from "../libs/Transition.sol";
+import { Offchain } from "./Offchain.sol";
+
+contract ClientFrontend {
+    using SafeMath for uint256;
+    using Tx for bytes;
+    using Types for Types.UserState;
+
+    function decodeTransfer(bytes calldata txBytes)
+        external
+        pure
+        returns (Offchain.Transfer memory _tx)
+    {
+        return Offchain.decodeTransfer(txBytes);
+    }
+
+    function decodeMassMigration(bytes calldata txBytes)
+        external
+        pure
+        returns (Offchain.MassMigration memory _tx)
+    {
+        return Offchain.decodeMassMigration(txBytes);
+    }
+
+    function decodeCreate2Transfer(bytes calldata txBytes)
+        external
+        pure
+        returns (Offchain.Create2Transfer memory _tx)
+    {
+        return Offchain.decodeCreate2Transfer(txBytes);
+    }
+
+    // Transfer
+
+    function processTransferTx(
+        bytes32 stateRoot,
+        bytes memory txBytes,
+        Types.StateMerkleProof memory from,
+        Types.StateMerkleProof memory to
+    )
+        public
+        pure
+        returns (
+            bytes32 newRoot,
+            bytes memory senderState,
+            bytes memory receiverState,
+            Types.Result result
+        )
+    {
+        Offchain.Transfer memory _tx = Offchain.decodeTransfer(txBytes);
+        uint256 tokenType = from.state.tokenType;
+        (newRoot, senderState, result) = Transition.processSender(
+            stateRoot,
+            _tx.fromIndex,
+            tokenType,
+            _tx.amount,
+            _tx.fee,
+            from
+        );
+        if (result != Types.Result.Ok)
+            return (newRoot, senderState, "", result);
+        (newRoot, receiverState, result) = Transition.processReceiver(
+            newRoot,
+            _tx.toIndex,
+            _tx.amount,
+            tokenType,
+            to
+        );
+        return (newRoot, senderState, receiverState, result);
+    }
+
+    function applyTransferTx(
```

Comment:
@vaibhavchellani  Does the go client require state root to be exposed in this API? [This line](https://github.com/thehubbleproject/reddit-commander/blob/ad827761d50f9b7989dc22286f18163ec858e204/core/tx.go#L92) suggests it doesn't, but not sure if I'm checking the right line

Reply:
Ideally I would want it to return the newState after apply, I am not sure about the purpose right now but I can see a usecase where a user would want `fast-sync`, and during that we can just apply transactions without processTx and just validate state updates via stateRoot check. 

But I think we can skip for now and add that later.